system-addon/lib/MessageManager.jsm

```diff
@@ -142,13 +142,16 @@ class MessageManager {
    * @param  {obj} msg.target A message target
    */
   onMessage(msg) {
-    const action = msg.data;
     const {portID} = msg.target;
-    if (!action || !action.type) {
+    if (!msg.data || !msg.data.type) {
       Cu.reportError(new Error(`Received an improperly formatted message from ${portID}`));
       return;
     }
-    this.onActionFromContent(action, msg.target.portID);
+    let action = {};
+    action.data = msg.data.data;
+    action.type = msg.data.type;
+    action.target = msg.target;
```

Comment:
can you use `Object.assign` (this this currently missing `meta`), and add a comment about target being a reference, so it should only be used in feeds, not reducers?